CMakeLists.txt

```diff
@@ -19,46 +19,29 @@ cmake_minimum_required(VERSION 3.12)
 
 project(vulkan_samples)
 
-# create output folder
-file(MAKE_DIRECTORY output)
-
 # Add path for local cmake scripts
 list(APPEND CMAKE_MODULE_PATH
     ${CMAKE_CURRENT_SOURCE_DIR}/bldsys/cmake
     ${CMAKE_CURRENT_SOURCE_DIR}/bldsys/cmake/module)
 
-# Precompiled header macro. Parameters are source file list and filename for pch cpp file.
-macro(vulkan_samples_pch SRCS PCHCPP)
-  if(MSVC AND CMAKE_GENERATOR MATCHES "^Visual Studio")
-    set(PCH_NAME "$(IntDir)\\pch.pch")
-    # make source files use/depend on PCH_NAME
-    set_source_files_properties(${${SRCS}} PROPERTIES COMPILE_FLAGS "/Yupch.h /FIpch.h /Fp${PCH_NAME} /Zm300" OBJECT_DEPENDS "${PCH_NAME}")
-    # make PCHCPP file compile and generate PCH_NAME
-    set_source_files_properties(${PCHCPP} PROPERTIES COMPILE_FLAGS "/Ycpch.h /Fp${PCH_NAME} /Zm300" OBJECT_OUTPUTS "${PCH_NAME}")
-    list(APPEND ${SRCS} "${PCHCPP}")
-  endif()
-endmacro(vulkan_samples_pch)
-
 include(utils)
-include(global_options)
+include(global_config)
 include(sample_helper)
 include(check_atomic)
+include(test_helpers)
+
+# options
+set(VKB_WARNINGS_AS_ERRORS ON CACHE BOOL "Enable Warnings as Errors")
+set(VKB_VALIDATION_LAYERS OFF CACHE BOOL "Enable validation layers for every application.")
+set(VKB_VALIDATION_LAYERS_GPU_ASSISTED OFF CACHE BOOL "Enable GPU assisted validation layers for every application.")
+set(VKB_VULKAN_DEBUG ON CACHE BOOL "Enable VK_EXT_debug_utils or VK_EXT_debug_marker if supported.")
+set(VKB_BUILD_SAMPLES ON CACHE BOOL "Enable generation and building of Vulkan best practice samples.")
+set(VKB_BUILD_TESTS OFF CACHE BOOL "Enable generation and building of Vulkan best practice tests.")
+set(VKB_WSI_SELECTION "XCB" CACHE STRING "Select WSI target (XCB, XLIB, WAYLAND, D2D)")
```

Comment:
As we're making changes to the location, think we can switch from set to option for optional variables?  The advantage is in cmake 3.13 and up option wont create a separate second variable.  Also, we can use dependent options.

Reply:
Sounds like a good edition. I prefer these variables in the main CMake file as it is easier to find all the options that you can use with a library. I expect in third parties for them to be in the main CMake, so it makes sense to follow this standard here

Reply:
I whole-heartedly agree for the same reasons you're saying.  One of the subtle objections I have with using set instead of option is it doesn't really denote what's safe for end users to change on a whim in 3rd party libraries.

Reply:
Should work with option now. I will let the CI decide